### PR TITLE
bpo-38291: Fix a spurious warning when using help(object)

### PR DESCRIPTION
**Pull request by srittau**
https://bugs.python.org/issue38291

Automerge-Triggered-By: GH:gvanrossum

### CODE REVIEW

**Comment by srittau** [comment]
This should not need a NEWS entry per:

>These are the two exceptions:
> 
> 1. [...]
> 2. If a change is a fix (or other adjustment) to an earlier unreleased change and the original NEWS entry remains valid, then no additional entry is needed.

**Comment by srittau** [comment]
The Azure build failure seems unrelated.

File: Lib/typing.py [hunk 1]
```diff
@@ -2512,7 +2512,7 @@ def __enter__(self) -> 'TextIO':
 
 class _DeprecatedType(type):
     def __getattribute__(cls, name):
-        if name != "__dict__" and name in cls.__dict__:
+        if name not in ["__dict__", "__module__"] and name in cls.__dict__:
```
Comment (gvanrossum) [inline Lib/typing.py]: If you use a tuple instead of a list, the compiler generates faster code. :-)

**Review by gvanrossum — APPROVED** [review]

**Comment by srittau** [comment]
This time, Azure passes, but the address sanitizer(?) test times out. Looks like an unrelated multiprocessing error, possibly a race condition.

**Comment by gvanrossum** [comment]
Alas, we have to close and reopen the PR to make the last test pass (hopefully), GitHub won't let me merge without that.

**Comment by gvanrossum** [comment]
Is a backport needed?

**Comment by srittau** [comment]
Thanks and no backport needed, the original change is just in the 3.11 branch.

**Pull request merged**